sympy/physics/continuum_mechanics/beam.py

```diff
@@ -2297,298 +2296,660 @@ def plot_ild_moment(self,subs=None):
                xlabel=r'$\mathrm{a}$', ylabel=r'$\mathrm{M}$', line_color='blue', show=True)
 
     @doctest_depends_on(modules=('numpy',))
-    def draw(self, pictorial=True):
-        """
-        Returns a plot object representing the beam diagram of the beam.
-        In particular, the diagram might include:
-
-        * the beam.
-        * vertical black arrows represent point loads and support reaction
-          forces (the latter if they have been added with the ``apply_load``
-          method).
-        * circular arrows represent moments.
-        * shaded areas represent distributed loads.
-        * the support, if ``apply_support`` has been executed.
-        * if a composite beam has been created with the ``join`` method and
-          a hinge has been specified, it will be shown with a white disc.
-
-        The diagram shows positive loads on the upper side of the beam,
-        and negative loads on the lower side. If two or more distributed
-        loads acts along the same direction over the same region, the
-        function will add them up together.
-
-        .. note::
-            The user must be careful while entering load values.
-            The draw function assumes a sign convention which is used
-            for plotting loads.
-            Given a right handed coordinate system with XYZ coordinates,
-            the beam's length is assumed to be along the positive X axis.
-            The draw function recognizes positive loads(with n>-2) as loads
-            acting along negative Y direction and positive moments acting
-            along positive Z direction.
+    def draw(self, show_load_values=True, draw_support_icons=True, show_grid=False, show_support_reactions=True):
```

Comment:
You can add new kwargs (possibly), but you cannot remove the `pictorial` kwarg because that is a backwards incompatible change.